business/statusService.js

```diff
@@ -109,6 +110,29 @@ class StatusService {
     })
   }
 
+  async _crawlbreakdown() {
+    const data = await requestPromise(
+      this._crawlerQuery(`
+      traces      
+      | where customDimensions.outcome == 'Processed'  
+      | where timestamp > ago(90d) 
+      | parse customDimensions.root with "component@cd:/" type "/" specTrail 
```

Comment:
there is a different dimension that we are more interested in here, "the tool" scancode vs licensee vs npm etc

```
| parse message with "Processed " tool "@cd:/" type "/" specTrail 
| summarize count() by when=bin(timestamp, 1d), tool, type
```
so there is a tool and a type dimension

![image](https://user-images.githubusercontent.com/5168796/58512116-22576e00-8151-11e9-8acf-773bebc34835.png)

This may change the way the graph has to be rendered. If we have to, we can drop the 'type' dimension, as the main goal of this query is to see the breakdown of the 'tool'

Reply:
This is what it's looking like now.

```json
[{"nuget":{"nuget":1},"licensee":{"nuget":1,"npm":23,"git":23},"npm":{"npm":23},"fossology":{"npm":23,"git":23},"scancode":{"npm":23,"git":23},"clearlydefined":{"git":23},"date":"2019-02-28T00:00:00Z"},{"licensee":{"git":1},"scancode":{"git":2},"date":"2019-03-01T00:00:00Z"},{"clearlydefined":{"git":1},"fossology":{"git":1},"licensee":{"git":1},"scancode":{"git":1},"date":"2019-03-04T00:00:00Z"},{"scancode":{"npm":2,"git":4},"licensee":{"npm":2,"git":4},"fossology":{"npm":2,"git":4},"npm":{"npm":2},"clearlydefined":{"git":4},"date":"2019-03-05T00:00:00Z"},{"npm":{"npm":1},"scancode":{"npm":2},"licensee":{"npm":1},"fossology":{"npm":1},"date":"2019-03-06T00:00:00Z"},{"scancode":{"pod":1},"pod":{"pod":1},"licensee":{"pod":1,"nuget":1},"fossology":{"pod":1},"nuget":{"nuget":1},"date":"2019-03-07T00:00:00Z"},{"fossology":{"npm":95,"git":29},"scancode":{"npm":95,"git":29},"npm":{"npm":96},"licensee":{"npm":95,"git":29},"clearlydefined":{"git":29},"date":"2019-03-08T00:00:00Z"},{"fossology":{"npm":1,"git":40},"scancode":{"npm":1,"git":42},"licensee":{"npm":1,"git":39},"clearlydefined":{"git":40},"date":"2019-03-09T00:00:00Z"},{"clearlydefined":{"sourcearchive":1,"git":80},"licensee":{"sourcearchive":1,"nuget":79,"npm":140,"git":74},"nuget":{"nuget":153},"scancode":{"npm":65,"git":13},"fossology":{"npm":70,"git":19},"npm":{"npm":158},"maven":{"maven":1},"date":"2019-03-12T00:00:00Z"},{"licensee":{"npm":23,"git":18},"fossology":{"npm":23,"git":19},"npm":{"npm":23},"scancode":{"npm":23,"git":18},"clearlydefined":{"git":18},"date":"2019-03-14T00:00:00Z"},{"scancode":{"sourcearchive":2,"pypi":1},"licensee":{"sourcearchive":3,"pypi":1},"fossology":{"sourcearchive":3,"pypi":1},"clearlydefined":{"sourcearchive":3},"pypi":{"pypi":1},"maven":{"maven":4},"date":"2019-03-18T00:00:00Z"},{"docker":{"docker":11},"date":"2019-03-19T00:00:00Z"},{"nuget":{"nuget":1},"licensee":{"nuget":1,"git":1},"scancode":{"git":1},"fossology":{"git":1},"clearlydefined":{"git":1},"docker":{"docker":37},"date":"2019-03-20T00:00:00Z"},{"clearlydefined":{"sourcearchive":2,"git":3},"fossology":{"sourcearchive":2,"npm":1,"git":3},"licensee":{"sourcearchive":2,"nuget":2,"npm":1,"git":3},"scancode":{"sourcearchive":2,"npm":1,"git":3},"nuget":{"nuget":3},"npm":{"npm":1},"maven":{"maven":2},"docker":{"docker":55},"date":"2019-03-21T00:00:00Z"},{"licensee":{"nuget":1,"npm":1,"git":2},"fossology":{"npm":1,"git":3},"npm":{"npm":1},"scancode":{"npm":1,"git":3},"clearlydefined":{"git":2},"date":"2019-03-22T00:00:00Z"},{"fossology":{"npm":495,"git":126},"npm":{"npm":494},"licensee":{"npm":493,"git":116},"scancode":{"npm":494,"git":127},"clearlydefined":{"git":125},"date":"2019-03-26T00:00:00Z"},{"fossology":{"git":224},"licensee":{"git":224},"clearlydefined":{"git":225},"scancode":{"git":223},"date":"2019-03-27T00:00:00Z"},{"nuget":{"nuget":211},"licensee":{"nuget":79,"npm":28,"git":27,"gem":3},"npm":{"npm":114},"fossology":{"npm":28,"git":26,"gem":2},"scancode":{"npm":28,"git":27,"gem":3},"clearlydefined":{"git":28},"gem":{"gem":3},"date":"2019-03-29T00:00:00Z"},{"nuget":{"nuget":1},"licensee":{"nuget":133,"npm":622,"git":247},"npm":{"npm":1110},"scancode":{"npm":620,"git":251},"fossology":{"npm":621,"git":259},"clearlydefined":{"git":259},"date":"2019-03-30T00:00:00Z"},{"scancode":{"npm":569,"git":119},"licensee":{"npm":568,"git":106},"fossology":{"npm":571,"git":118},"clearlydefined":{"git":117},"date":"2019-03-31T00:00:00Z"},{"fossology":{"npm":28,"git":142,"gem":1},"licensee":{"npm":28,"git":141,"gem":1},"npm":{"npm":124},"scancode":{"npm":28,"git":141,"gem":1},"clearlydefined":{"git":141},"gem":{"gem":1},"date":"2019-04-01T00:00:00Z"},{"clearlydefined":{"sourcearchive":1,"git":197},"licensee":{"sourcearchive":1,"nuget":299,"npm":340,"git":191},"fossology":{"sourcearchive":1,"npm":340,"git":197},"scancode":{"sourcearchive":2,"npm":340,"git":195},"nuget":{"nuget":601},"npm":{"npm":611},"maven":{"maven":12},"date":"2019-04-02T00:00:00Z"},{"npm":{"npm":369},"fossology":{"npm":369,"git":223},"licensee":{"npm":370,"git":218},"scancode":{"npm":371,"git":221},"clearlydefined":{"git":224},"date":"2019-04-03T00:00:00Z"},{"clearlydefined":{"sourcearchive":1,"git":119},"licensee":{"sourcearchive":1,"pypi":5,"nuget":19,"npm":94,"git":118,"crate":1},"scancode":{"pypi":5,"npm":94,"git":121,"crate":1},"fossology":{"pypi":5,"npm":94,"git":119,"crate":1},"pypi":{"pypi":17},"nuget":{"nuget":53},"npm":{"npm":140},"maven":{"maven":11},"crate":{"crate":4},"date":"2019-04-04T00:00:00Z"},{"fossology":{"sourcearchive":12,"pypi":36,"npm":153,"git":75,"crate":7},"licensee":{"sourcearchive":13,"pypi":36,"nuget":80,"npm":152,"git":70,"crate":7},"scancode":{"sourcearchive":12,"pypi":36,"npm":153,"git":74,"crate":7},"clearlydefined":{"sourcearchive":13,"git":74},"pypi":{"pypi":114},"nuget":{"nuget":54},"npm":{"npm":138},"maven":{"maven":20},"crate":{"crate":5},"date":"2019-04-05T00:00:00Z"},{"scancode":{"sourcearchive":14,"pypi":7,"npm":31,"git":41,"crate":1},"fossology":{"sourcearchive":14,"pypi":7,"npm":31,"git":42,"crate":1},"licensee":{"sourcearchive":14,"pypi":7,"nuget":10,"npm":30,"git":40,"crate":1},"clearlydefined":{"sourcearchive":14,"git":42},"date":"2019-04-06T00:00:00Z"},{"licensee":{"sourcearchive":5,"pypi":5,"nuget":29,"npm":69,"git":28,"crate":13},"scancode":{"sourcearchive":5,"pypi":5,"npm":70,"git":31,"crate":13},"fossology":{"sourcearchive":5,"pypi":5,"npm":71,"git":30,"crate":13},"clearlydefined":{"sourcearchive":5,"git":30},"pypi":{"pypi":33},"nuget":{"nuget":61},"npm":{"npm":105},"maven":{"maven":13},"crate":{"crate":13},"date":"2019-04-07T00:00:00Z"},{"scancode":{"sourcearchive":21,"pypi":3,"npm":147,"git":88},"licensee":{"sourcearchive":23,"pypi":3,"nuget":126,"npm":145,"git":90},"clearlydefined":{"sourcearchive":23,"git":91},"fossology":{"sourcearchive":21,"pypi":3,"npm":144,"git":90},"pypi":{"pypi":16},"nuget":{"nuget":94},"npm":{"npm":110},"maven":{"maven":14},"date":"2019-04-08T00:00:00Z"},{"fossology":{"sourcearchive":148,"pypi":5,"npm":28,"git":19},"licensee":{"sourcearchive":148,"pypi":5,"nuget":10,"npm":27,"git":18},"scancode":{"sourcearchive":148,"pypi":5,"npm":28,"git":19},"clearlydefined":{"sourcearchive":147,"git":19},"pypi":{"pypi":17},"nuget":{"nuget":10},"npm":{"npm":28},"maven":{"maven":150},"date":"2019-04-09T00:00:00Z"},{"scancode":{"sourcearchive":84,"pypi":4,"npm":36,"git":20},"clearlydefined":{"sourcearchive":84,"git":22},"licensee":{"sourcearchive":84,"pypi":4,"nuget":20,"npm":35,"git":20},"fossology":{"sourcearchive":84,"pypi":4,"npm":36,"git":22},"pypi":{"pypi":19},"nuget":{"nuget":20},"npm":{"npm":36},"maven":{"maven":86},"date":"2019-04-10T00:00:00Z"},{"scancode":{"sourcearchive":13,"pypi":15,"npm":39,"git":28,"crate":6},"fossology":{"sourcearchive":12,"pypi":15,"npm":40,"git":28,"crate":6},"licensee":{"sourcearchive":12,"pypi":15,"nuget":17,"npm":39,"git":28,"crate":6},"clearlydefined":{"sourcearchive":12,"git":28},"pypi":{"pypi":28},"nuget":{"nuget":17},"npm":{"npm":39},"maven":{"maven":12},"crate":{"crate":6},"date":"2019-04-11T00:00:00Z"},{"licensee":{"sourcearchive":15,"pypi":5,"nuget":13,"npm":36,"git":17},"clearlydefined":{"sourcearchive":15,"git":18},"scancode":{"sourcearchive":15,"pypi":5,"npm":36,"git":18},"fossology":{"sourcearchive":15,"pypi":5,"npm":36,"git":18},"pypi":{"pypi":21},"nuget":{"nuget":13},"npm":{"npm":36},"maven":{"maven":16},"date":"2019-04-12T00:00:00Z"},{"fossology":{"sourcearchive":40,"pypi":27,"npm":151,"git":120,"crate":4},"scancode":{"sourcearchive":40,"pypi":27,"npm":151,"git":119,"crate":4},"clearlydefined":{"sourcearchive":40,"git":120},"licensee":{"sourcearchive":40,"pypi":27,"nuget":94,"npm":150,"git":113,"crate":4},"pypi":{"pypi":33},"nuget":{"nuget":94},"npm":{"npm":151},"maven":{"maven":47},"crate":{"crate":4},"date":"2019-04-13T00:00:00Z"},{"scancode":{"sourcearchive":4,"pypi":17,"npm":58,"git":38,"crate":3},"clearlydefined":{"sourcearchive":4,"git":39},"licensee":{"sourcearchive":4,"pypi":17,"nuget":31,"npm":59,"git":38,"crate":3},"fossology":{"sourcearchive":4,"pypi":17,"npm":59,"git":39,"crate":3},"pypi":{"pypi":28},"nuget":{"nuget":31},"npm":{"npm":59},"maven":{"maven":4},"crate":{"crate":3},"date":"2019-04-14T00:00:00Z"},{"nuget":{"nuget":23},"licensee":{"nuget":23,"npm":176,"git":98,"gem":1},"npm":{"npm":176},"scancode":{"npm":176,"git":99,"gem":1},"fossology":{"npm":176,"git":105,"gem":1},"clearlydefined":{"git":104},"gem":{"gem":1},"date":"2019-04-15T00:00:00Z"},{"clearlydefined":{"sourcearchive":7,"git":23},"fossology":{"sourcearchive":7,"pypi":8,"npm":30,"git":23,"crate":1},"licensee":{"sourcearchive":7,"pypi":8,"nuget":9,"npm":30,"git":23,"crate":1},"scancode":{"sourcearchive":7,"pypi":8,"npm":30,"git":23,"crate":1},"pypi":{"pypi":19},"nuget":{"nuget":9},"npm":{"npm":30},"maven":{"maven":7},"crate":{"crate":1},"date":"2019-04-16T00:00:00Z"},{"licensee":{"sourcearchive":22,"pypi":19,"nuget":36,"npm":87,"git":50,"crate":2},"fossology":{"sourcearchive":22,"pypi":19,"npm":87,"git":50,"crate":2},"scancode":{"sourcearchive":22,"pypi":19,"npm":87,"git":49,"crate":2},"clearlydefined":{"sourcearchive":22,"git":50},"pypi":{"pypi":24},"nuget":{"nuget":36},"npm":{"npm":87},"maven":{"maven":23},"crate":{"crate":2},"date":"2019-04-17T00:00:00Z"},{"fossology":{"sourcearchive":9,"pypi":4,"npm":46,"git":30,"crate":2},"scancode":{"sourcearchive":9,"pypi":4,"npm":46,"git":29,"crate":2},"clearlydefined":{"sourcearchive":9,"git":30},"licensee":{"sourcearchive":9,"pypi":4,"nuget":12,"npm":46,"git":30,"crate":2},"pypi":{"pypi":14},"nuget":{"nuget":12},"npm":{"npm":46},"maven":{"maven":10},"crate":{"crate":2},"date":"2019-04-18T00:00:00Z"},{"scancode":{"sourcearchive":7,"pypi":4,"npm":58,"git":29,"crate":5},"licensee":{"sourcearchive":7,"pypi":4,"nuget":17,"npm":58,"git":29,"crate":5},"clearlydefined":{"sourcearchive":7,"git":29},"fossology":{"sourcearchive":7,"pypi":4,"npm":58,"git":29,"crate":5},"pypi":{"pypi":10},"nuget":{"nuget":17},"npm":{"npm":58},"maven":{"maven":7},"crate":{"crate":5},"date":"2019-04-19T00:00:00Z"},{"scancode":{"sourcearchive":6,"pypi":3,"npm":23,"git":14},"clearlydefined":{"sourcearchive":6,"git":14},"fossology":{"sourcearchive":6,"pypi":3,"npm":23,"git":14},"licensee":{"sourcearchive":6,"pypi":3,"nuget":11,"npm":23,"git":14},"pypi":{"pypi":11},"nuget":{"nuget":11},"npm":{"npm":23},"maven":{"maven":6},"date":"2019-04-20T00:00:00Z"},{"scancode":{"sourcearchive":4,"pypi":5,"npm":53,"git":22,"crate":1},"fossology":{"sourcearchive":4,"pypi":5,"npm":53,"git":23,"crate":1},"clearlydefined":{"sourcearchive":4,"git":23},"licensee":{"sourcearchive":4,"pypi":5,"nuget":21,"npm":53,"git":23,"crate":1},"pypi":{"pypi":16},"nuget":{"nuget":21},"npm":{"npm":53},"maven":{"maven":4},"crate":{"crate":1},"date":"2019-04-21T00:00:00Z"},{"scancode":{"sourcearchive":4,"pypi":1,"npm":46,"git":21,"crate":3},"fossology":{"sourcearchive":4,"pypi":1,"npm":46,"git":22,"crate":3},"licensee":{"sourcearchive":4,"pypi":1,"nuget":5,"npm":46,"git":23,"crate":3},"clearlydefined":{"sourcearchive":4,"git":22},"pypi":{"pypi":14},"nuget":{"nuget":5},"npm":{"npm":46},"maven":{"maven":4},"crate":{"crate":3},"date":"2019-04-22T00:00:00Z"},{"scancode":{"sourcearchive":2,"npm":17,"maven":3,"git":7,"gem":1},"fossology":{"sourcearchive":2,"npm":17,"maven":3,"git":7,"gem":1},"clearlydefined":{"sourcearchive":2,"git":7},"licensee":{"sourcearchive":2,"nuget":4,"npm":17,"maven":3,"git":7,"gem":1},"pypi":{"pypi":13},"nuget":{"nuget":4},"npm":{"npm":17},"maven":{"maven":5},"gem":{"gem":1},"date":"2019-04-23T00:00:00Z"},{"licensee":{"sourcearchive":12,"pypi":7,"nuget":20,"npm":108,"maven":13,"git":37,"crate":3},"clearlydefined":{"sourcearchive":12,"git":37},"fossology":{"sourcearchive":12,"pypi":7,"npm":108,"maven":13,"git":37,"crate":3},"scancode":{"sourcearchive":12,"pypi":7,"npm":108,"maven":12,"git":34,"crate":3},"pypi":{"pypi":17},"nuget":{"nuget":20},"npm":{"npm":108},"maven":{"maven":13},"crate":{"crate":3},"date":"2019-04-24T00:00:00Z"},{"licensee":{"sourcearchive":9,"pypi":5,"nuget":12,"npm":60,"maven":9,"git":29,"crate":1},"fossology":{"sourcearchive":9,"pypi":5,"npm":60,"maven":9,"git":29,"crate":1},"scancode":{"sourcearchive":9,"pypi":5,"npm":60,"maven":10,"git":29,"crate":1},"clearlydefined":{"sourcearchive":9,"git":29},"pypi":{"pypi":18},"nuget":{"nuget":12},"npm":{"npm":60},"maven":{"maven":9},"crate":{"crate":1},"date":"2019-04-25T00:00:00Z"},{"clearlydefined":{"sourcearchive":1,"git":32},"scancode":{"sourcearchive":1,"pypi":4,"npm":55,"maven":1,"git":32},"fossology":{"sourcearchive":1,"pypi":4,"npm":55,"maven":1,"git":32},"licensee":{"sourcearchive":1,"pypi":4,"nuget":4,"npm":55,"maven":1,"git":32},"pypi":{"pypi":18},"nuget":{"nuget":4},"npm":{"npm":55},"maven":{"maven":1},"date":"2019-04-26T00:00:00Z"},{"licensee":{"sourcearchive":1,"pypi":1,"nuget":11,"npm":51,"maven":1,"git":29,"crate":1},"fossology":{"sourcearchive":1,"pypi":1,"npm":51,"maven":1,"git":29,"crate":1},"clearlydefined":{"sourcearchive":1,"git":29},"scancode":{"sourcearchive":1,"pypi":1,"npm":51,"maven":1,"git":29,"crate":1},"pypi":{"pypi":14},"nuget":{"nuget":11},"npm":{"npm":51},"maven":{"maven":1},"crate":{"crate":1},"date":"2019-04-27T00:00:00Z"},{"scancode":{"pypi":10,"npm":54,"git":20,"crate":1},"pypi":{"pypi":19},"licensee":{"pypi":10,"nuget":8,"npm":54,"git":21,"crate":1},"fossology":{"pypi":10,"npm":54,"git":21,"crate":1},"nuget":{"nuget":8},"npm":{"npm":54},"clearlydefined":{"git":21},"crate":{"crate":1},"date":"2019-04-28T00:00:00Z"},{"clearlydefined":{"sourcearchive":8,"git":10},"licensee":{"sourcearchive":8,"pypi":2,"nuget":3,"npm":28,"maven":9,"git":10},"scancode":{"sourcearchive":8,"pypi":2,"npm":28,"maven":9,"git":10},"fossology":{"sourcearchive":8,"pypi":2,"npm":28,"maven":9,"git":10},"pypi":{"pypi":12},"nuget":{"nuget":3},"npm":{"npm":28},"maven":{"maven":9},"date":"2019-04-29T00:00:00Z"},{"licensee":{"sourcearchive":9,"pypi":1,"nuget":4,"npm":12,"maven":10,"git":5},"clearlydefined":{"sourcearchive":9,"git":5},"scancode":{"sourcearchive":9,"pypi":2,"npm":11,"maven":9,"git":4},"fossology":{"sourcearchive":9,"pypi":1,"npm":12,"maven":9,"git":5},"pypi":{"pypi":14},"nuget":{"nuget":4},"npm":{"npm":12},"maven":{"maven":9},"date":"2019-04-30T00:00:00Z"},{"nuget":{"nuget":22},"scancode":{"npm":82,"git":67},"npm":{"npm":82},"clearlydefined":{"git":64},"date":"2019-05-01T00:00:00Z"},{"scancode":{"sourcearchive":1,"pypi":5,"npm":42,"maven":1,"git":22,"gem":1},"clearlydefined":{"sourcearchive":1,"git":25},"pypi":{"pypi":16},"nuget":{"nuget":14},"npm":{"npm":42},"maven":{"maven":1},"gem":{"gem":1},"date":"2019-05-02T00:00:00Z"},{"scancode":{"pypi":2,"npm":10,"git":1},"pypi":{"pypi":15},"npm":{"npm":10},"clearlydefined":{"git":1},"date":"2019-05-03T00:00:00Z"},{"pypi":{"pypi":15},"scancode":{"pypi":6,"npm":20,"git":20,"crate":4},"nuget":{"nuget":11},"npm":{"npm":20},"clearlydefined":{"git":20},"crate":{"crate":4},"date":"2019-05-04T00:00:00Z"},{"scancode":{"pypi":3,"npm":32,"git":17},"pypi":{"pypi":18},"nuget":{"nuget":6},"npm":{"npm":32},"clearlydefined":{"git":17},"date":"2019-05-05T00:00:00Z"},{"scancode":{"sourcearchive":7,"pypi":14,"npm":87,"maven":7,"git":51,"crate":4},"clearlydefined":{"sourcearchive":7,"git":50},"pypi":{"pypi":29},"nuget":{"nuget":15},"npm":{"npm":87},"maven":{"maven":7},"crate":{"crate":4},"date":"2019-05-06T00:00:00Z"},{"scancode":{"sourcearchive":1,"pypi":6,"npm":33,"maven":1,"git":25,"crate":5},"clearlydefined":{"sourcearchive":1,"git":25},"pypi":{"pypi":20},"nuget":{"nuget":9},"npm":{"npm":33},"maven":{"maven":1},"crate":{"crate":5},"date":"2019-05-07T00:00:00Z"},{"scancode":{"sourcearchive":4,"pypi":1,"npm":18,"maven":4,"git":6},"clearlydefined":{"sourcearchive":4,"git":6},"pypi":{"pypi":19},"nuget":{"nuget":3},"npm":{"npm":18},"maven":{"maven":4},"date":"2019-05-08T00:00:00Z"},{"clearlydefined":{"sourcearchive":1,"git":2},"scancode":{"sourcearchive":1,"npm":6,"maven":1,"git":2},"pypi":{"pypi":17},"nuget":{"nuget":2},"npm":{"npm":6},"maven":{"maven":1},"date":"2019-05-09T00:00:00Z"},{"clearlydefined":{"sourcearchive":2,"git":10},"scancode":{"sourcearchive":2,"pypi":2,"npm":21,"maven":3,"git":10},"pypi":{"pypi":23},"nuget":{"nuget":12},"npm":{"npm":21},"maven":{"maven":3},"date":"2019-05-10T00:00:00Z"},{"clearlydefined":{"sourcearchive":5,"git":33},"scancode":{"sourcearchive":5,"pypi":13,"npm":59,"maven":5,"git":32},"pypi":{"pypi":31},"nuget":{"nuget":14},"npm":{"npm":59},"maven":{"maven":5},"date":"2019-05-11T00:00:00Z"},{"scancode":{"sourcearchive":9,"pypi":32,"npm":76,"maven":19,"git":39,"crate":1},"clearlydefined":{"sourcearchive":9,"git":41},"pypi":{"pypi":47},"nuget":{"nuget":23},"npm":{"npm":76},"maven":{"maven":19},"crate":{"crate":1},"date":"2019-05-12T00:00:00Z"},{"clearlydefined":{"sourcearchive":2,"git":12},"scancode":{"sourcearchive":2,"pypi":4,"npm":24,"maven":3,"git":12},"pypi":{"pypi":21},"nuget":{"nuget":7},"npm":{"npm":24},"maven":{"maven":3},"date":"2019-05-13T00:00:00Z"},{"scancode":{"sourcearchive":5,"npm":11,"maven":5,"git":8},"clearlydefined":{"sourcearchive":5,"git":8},"pypi":{"pypi":42},"nuget":{"nuget":4},"npm":{"npm":11},"maven":{"maven":5},"date":"2019-05-14T00:00:00Z"},{"scancode":{"sourcearchive":1,"pypi":18,"npm":27,"git":25},"clearlydefined":{"sourcearchive":1,"git":24},"pypi":{"pypi":78},"nuget":{"nuget":8},"npm":{"npm":27},"date":"2019-05-15T00:00:00Z"},{"scancode":{"pypi":6,"npm":80,"git":39},"pypi":{"pypi":80},"npm":{"npm":80},"clearlydefined":{"git":39},"date":"2019-05-16T00:00:00Z"},{"clearlydefined":{"sourcearchive":13,"git":104},"scancode":{"sourcearchive":13,"pypi":12,"npm":124,"maven":13,"git":104},"pypi":{"pypi":31},"nuget":{"nuget":45},"npm":{"npm":124},"maven":{"maven":13},"date":"2019-05-17T00:00:00Z"},{"clearlydefined":{"sourcearchive":36,"git":58},"scancode":{"sourcearchive":36,"pypi":31,"npm":202,"maven":69,"git":59},"pypi":{"pypi":67},"nuget":{"nuget":61},"npm":{"npm":202},"maven":{"maven":69},"date":"2019-05-18T00:00:00Z"},{"scancode":{"sourcearchive":71,"pypi":13,"npm":150,"maven":49,"git":107},"clearlydefined":{"sourcearchive":71,"git":109},"pypi":{"pypi":93},"nuget":{"nuget":34},"npm":{"npm":150},"maven":{"maven":49},"date":"2019-05-19T00:00:00Z"},{"clearlydefined":{"sourcearchive":9,"git":36},"scancode":{"sourcearchive":9,"pypi":7,"npm":56,"maven":4,"git":36},"pypi":{"pypi":94},"nuget":{"nuget":10},"npm":{"npm":58},"maven":{"maven":4},"date":"2019-05-20T00:00:00Z"},{"pypi":{"pypi":78},"scancode":{"pypi":4,"npm":28,"git":18},"nuget":{"nuget":9},"npm":{"npm":43},"clearlydefined":{"git":21},"date":"2019-05-21T00:00:00Z"},{"npm":{"npm":169},"scancode":{"npm":25,"git":40},"clearlydefined":{"git":39},"date":"2019-05-22T00:00:00Z"},{"licensee":{"sourcearchive":44,"pypi":45,"nuget":27,"npm":71,"maven":47,"git":62},"scancode":{"sourcearchive":44,"pypi":25,"npm":72,"maven":46,"git":48},"clearlydefined":{"sourcearchive":44,"git":49},"pypi":{"pypi":65},"nuget":{"nuget":26},"npm":{"npm":71},"maven":{"maven":45},"date":"2019-05-23T00:00:00Z"},{"licensee":{"sourcearchive":15,"pypi":32,"nuget":27,"npm":104,"maven":16,"git":88},"clearlydefined":{"sourcearchive":15,"git":81},"scancode":{"sourcearchive":15,"pypi":19,"npm":104,"maven":16,"git":81},"pypi":{"pypi":77},"nuget":{"nuget":27},"npm":{"npm":104},"maven":{"maven":16},"date":"2019-05-24T00:00:00Z"},{"licensee":{"pypi":17,"nuget":36,"npm":73,"git":22},"pypi":{"pypi":43},"scancode":{"pypi":13,"npm":73,"git":19},"nuget":{"nuget":36},"npm":{"npm":73},"clearlydefined":{"git":20},"date":"2019-05-25T00:00:00Z"},{"scancode":{"sourcearchive":12,"pypi":8,"npm":104,"maven":12,"git":84},"clearlydefined":{"sourcearchive":12,"git":82},"licensee":{"sourcearchive":12,"pypi":10,"nuget":14,"npm":104,"maven":12,"git":87},"pypi":{"pypi":108},"nuget":{"nuget":14},"npm":{"npm":104},"maven":{"maven":12},"date":"2019-05-26T00:00:00Z"},{"pypi":{"pypi":40},"licensee":{"pypi":1,"nuget":4,"npm":22,"git":6},"nuget":{"nuget":9},"npm":{"npm":56},"scancode":{"npm":23,"git":6},"clearlydefined":{"git":6},"date":"2019-05-29T00:00:00Z"}]
```

Reply:
Please @storrisi take a look at the new shape of the data, in case changes are needed on the website. Thank you.

Reply:
Thanks @ignacionr , will double check this

Reply:
With this kind of structure now we have several values for the same object:

```
"licensee":{"nuget":1,"npm":23,"git":23},"nuget":{"nuget":1},"npm":{"npm":23},"scancode":{"npm":23,"git":23},"fossology":{"npm":23,"git":23},"clearlydefined":{"git":23},"date":"2019-02-28T00:00:00Z"}
```

This means that the graph should be rendered using the tool has a key: "licensee", "scancode" an so on.
But what about the values of each tool? How should we render each property if those are other nested values?

Reply:
We could have almost 12 tools per date, so i think it would become very unreadable by the user.